### PR TITLE
bpo-37616: Enhance calculate_zip_path()

### PR DESCRIPTION
**Pull request by vstinner**
Use sprintf() to format the filename, rather than accessing directly
characters of the VERSION string.

Make getpath.c ready for Python version 3.100 or version 10.0.

https://bugs.python.org/issue37616

### CODE REVIEW

**Comment by vstinner** [comment]
@pablogsal proposed PR #20214.

**Comment by pablogsal** [comment]
This solution is great but lacks preprocessor magic 😆

File: Modules/getpath.c [hunk 1]
```diff
@@ -1296,16 +1296,28 @@ calculate_zip_path(PyCalculatePath *calculate)
 {
     PyStatus res;
 
+    // wsprintf() is not available on all platforms:
```
Comment (pablogsal) [inline Modules/getpath.c:1299]: I honestly find this code harder to follow and more complex than the preprocessor solution

**Comment by vstinner** [comment]
@pablogsal's PR #20214 is way simpler (and wa merged). I close my PR.

**Pull request closed without merging**